lessons/fast-track/filesystem/index.md

```diff
@@ -0,0 +1,198 @@
+podobně jako se z písmenek skládá text.
+
+Soubory jsou většinou uloženy na disku (nebo podobném médiu), kde bývá místo
+na bilióny bajtů.
+Aby počítač poznal, kde na disku je který soubor, používá *souborový systém*
+(angl. *filesystem*).
+Ten plní podobnou funkci jako v knížce obsah, jména kapitol a čísla stránek.
+
+
+## Operační systémy
+
+Souborových systémů existuje spousta druhů.
+Experti nám snad prominou hrubé zjednodušení, když si je rozdělíme na dva
+druhy: ty pro Windows a ty pro Unix.
+
+Unix je operační systém, vytvořený v sedmdesátých letech, ze kterého vycházejí
+dnešní systémy Linux, macOS a další.
+Základní principy, o kterých bude řeč tady, se od dob Unixu většinou
+příliš nezměnily.
+A tak když v těchto materiákech uvidíš jméno „Unix“, jde o něco společné pro
+Linux i macOS.
+Hlavní rozdíly mezi Linuxem a macOS jsou v konvencích – např. na Linuxu se
+místo pro domovské adresáře jmenuje většinou `/home`, kdežto na macOS `/Users`.
+
+Další rozšířený operační systém, Windows, z Unixu nevychází.
+Některé věci se v něm, jak později uvidíme, chovají jinak.
+
+
+## Adresáře
+
+Na dnešních souborových systémech jsou soubory tříděny do *adresářů* neboli
+*složek* (angl. *directory*, *folder*).
+Adresář může obsahovat spoustu souborů nebo i jiných adresářů.
+
+A teď něco, co pro tebe může být nové: pro programátory jsou adresáře taky soubory.
+Souborů je dokonce spousta druhů: *normální soubory* s informacemi, adresáře
+(které obsahují další soubory), speciální soubory které můžou reprezentovat
+celý disk nebo spojení mezi počítači, odkazy na jiné soubory, a tak dále.
+Co je a co není soubor závisí na systému.
+Dnes se proto omezíme jen na dva druhy souborů, které najdeme jak na Windows
+tak na Unixu: normální datové soubory (ty, které si pod jménem „soubor“
+představí běžný uživatel) a adresáře.
+
+
+## Cesty
+
+Abys mohl/a najít nějaký soubor, potřebuješ znát jeho jméno a adresář,
+který ten soubor obsahuje.
+Abys pak mohl/a najít ten adresář, musíš opět znát jméno adresáře a adresář,
+který ho obsahuje.
+A tak dál, až se dostaneš ke *kořenovému adresáři* (angl. *root directory*),
+který (zjednodušeně řečeno) obsahuje celý souborový systém.
+Když napíšeš jména všech adresářů které takhle projdeš za sebe, dostaneš
+*cestu* (angl. *path*) k danému souboru.
+Taková cesta by mohla na Unixu být třeba:
+
+* Linux: `/home/janca/Documents/archiv.tar.gz`
+* macOS: `/Users/janca/Documents/archiv.tar.gz`
+
+To znamená, že začneš v kořenovém adresáři (který se na Linuxu jmenuje `/`),
+v něm hledáš adresář `home` nebo `Users` (ten tradičně obsahuje domovské
+adresáře uživatelů), v něm pak `janca` (podle uživatelského jména),
+v něm `Documents`, a v něm pak `archiv.tar.gz`.
+To už není adresář, ale normální soubor do kterého se dají zapsat informace.
+
+Obdobná cesta na Windows by mohla být třeba:
+`C:\Users\Jana\Documents\archiv.tar.gz`
+
+Tahle cesta začíná na disku `C:`.
+Windows mají na rozdíl od Unixu zvláštní souborový systém pro každý disk,
```

Comment:
Řekl bych, že všechny systémy (Unix/Win) mají samostatný/zvláštní/oddělený souborový systém pro každý disk. Tedy kromě speciálních FS, které se dají natáhnout přes více fyzických disků.

Bez zabíhání do detailů bych řekl, že to hlavní k vysvětlení tady je, že na Windows má každý disk/FS vlastní kořenový adresář a na Unixu je ten hlavní opravu jen jeden.

Co třeba jen: Windows mají na rozdíl od Unixu zvláštní kořenový adresář pro každý disk – třeba `C:\` a `D:\` – zatímco na Unixu je hlavní kořenový adresář jen jeden a disky jsou v něm připojené jako speciální adresáře.